src/main/java/org/candlepin/subscriptions/event/EventController.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.event;
+
+import org.candlepin.subscriptions.db.EventRecordRepository;
+import org.candlepin.subscriptions.db.model.EventRecord;
+import org.candlepin.subscriptions.json.Event;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import org.springframework.stereotype.Controller;
+
+import java.time.OffsetDateTime;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import javax.persistence.EntityNotFoundException;
+import javax.transaction.Transactional;
+
+/**
+ * Encapsulates interaction with event store.
+ */
+@Controller
+public class EventController {
+    private final EventRecordRepository repo;
+
+    public EventController(EventRecordRepository repo) {
+        this.repo = repo;
+    }
+
+    /**
+     * Note: calling method needs to use @Transactional
+     *
+     * @param accountNumber account identifier
+     * @param begin beginning of the time range (inclusive)
+     * @param end end of the time range (exclusive)
+     * @return stream of Event
+     */
+    public Stream<Event> fetchEventsInTimeRange(String accountNumber, OffsetDateTime begin,
+        OffsetDateTime end) {
+        return repo
+            .findByAccountNumberAndTimestampGreaterThanEqualAndTimestampLessThanOrderByTimestamp(
+            accountNumber, begin, end).map(EventRecord::getEvent);
```

Comment:
Is this repo method used anywhere else/does it make sense to use projections to have the method return `Event` instead?

Reply:
Good idea. I briefly looked into projections, but initially gave up early on, but I will revisit and let you know if I have any luck.

Reply:
I had no luck. The projection idea would work well for having a subset of fields (like a DB view), but not so much for transforming a column. We can revisit if we find a nice way to implement, but for now, otherwise, tried to encapsulate all bits into that EventController.